transports/quic/src/provider/tokio.rs

```diff
@@ -18,68 +18,26 @@
 // FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
 // DEALINGS IN THE SOFTWARE.
 
-use futures::{future::BoxFuture, ready, Future, FutureExt};
+use futures::{future::BoxFuture, Future, FutureExt};
 use std::{
     io,
-    net::SocketAddr,
     task::{Context, Poll},
     time::Duration,
 };
-use tokio::{io::ReadBuf, net::UdpSocket};
 
 use crate::GenTransport;
 
 /// Transport with [`tokio`] runtime.
 pub type Transport = GenTransport<Provider>;
 
-/// Provider for reading / writing to a sockets and spawning
-/// tasks using [`tokio`].
-pub struct Provider {
-    socket: UdpSocket,
-    socket_recv_buffer: Vec<u8>,
-    next_packet_out: Option<(Vec<u8>, SocketAddr)>,
-}
+/// Provider for quinn runtime and spawning tasks using [`tokio`].
+pub struct Provider;
 
 impl super::Provider for Provider {
     type IfWatcher = if_watch::tokio::IfWatcher;
 
-    fn from_socket(socket: std::net::UdpSocket) -> std::io::Result<Self> {
-        let socket = UdpSocket::from_std(socket)?;
-        Ok(Provider {
-            socket,
-            socket_recv_buffer: vec![0; super::RECEIVE_BUFFER_SIZE],
-            next_packet_out: None,
-        })
-    }
-
-    fn poll_send_flush(&mut self, cx: &mut Context<'_>) -> Poll<Result<(), io::Error>> {
-        let (data, addr) = match self.next_packet_out.as_ref() {
-            Some(pending) => pending,
-            None => return Poll::Ready(Ok(())),
-        };
-        match self.socket.poll_send_to(cx, data.as_slice(), *addr) {
-            Poll::Ready(result) => {
-                self.next_packet_out = None;
-                Poll::Ready(result.map(|_| ()))
-            }
-            Poll::Pending => Poll::Pending,
-        }
-    }
-
-    fn poll_recv_from(&mut self, cx: &mut Context<'_>) -> Poll<io::Result<(Vec<u8>, SocketAddr)>> {
-        let Self {
-            socket,
-            socket_recv_buffer,
-            ..
-        } = self;
-        let mut read_buf = ReadBuf::new(socket_recv_buffer.as_mut_slice());
-        let packet_src = ready!(socket.poll_recv_from(cx, &mut read_buf)?);
-        let bytes = read_buf.filled().to_vec();
-        Poll::Ready(Ok((bytes, packet_src)))
-    }
-
-    fn start_send(&mut self, data: Vec<u8>, addr: SocketAddr) {
-        self.next_packet_out = Some((data, addr));
+    fn runtime() -> super::Runtime {
+        super::Runtime::Tokio
     }
 
     fn spawn(future: impl Future<Output = ()> + Send + 'static) {
```

Comment:
This is now only used in tests I think. Can we remove it from the prod-code then?

Reply:
Sorry, that's not possible: https://users.rust-lang.org/t/what-are-the-rules-for-cfg-test/54122/2

Reply:
@thomaseizinger fixed with 7505c6569147e5ff42f276b93b3b16f9b0dd189f. @kpp hope you don't mind me pushing a commit.